sdk/src/main/java/io/opentdf/platform/sdk/nanotdf/NanoTDFType.java

```diff
@@ -0,0 +1,43 @@
+package io.opentdf.platform.sdk.nanotdf;
+
+public class NanoTDFType {
+    public enum ECCurve {
+        SECP256R1("secp256r1"),
+        SECP384R1("secp384r1"),
+        SECP521R1("secp384r1"),
+        SECP256K1("secp256k1");
+
+        private String name;
+
+        ECCurve(String curveName) {
```

Comment:
Should have argument validation here - throw exception if the string does not match one of the enum values